### PR TITLE
3926: Public API Order Keyword Search

### PR DESCRIPTION
**Pull request by sutt0n**
This also moves functionality for `orderAdvancedSearchInteractor` into a use-case helper to be shared between an internal interactor and public interactor (since both share the same search functionality for now)

### CODE REVIEW

**Review by mrinsin — APPROVED** [review]

File: shared/src/business/useCaseHelper/search/orderKeywordSearch.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    .getSearchClient()
+    .search({
+      body: orderQuery,
+      index: 'efcms',
+    });
+
+  const hits = get(orderQueryMatchesBody, 'hits.hits', []);
+  const foundOrders = hits.map(hit =>
+    AWS.DynamoDB.Converter.unmarshall(hit['_source']),
+  );
```
Comment (kkoskelin) [inline shared/src/business/useCaseHelper/search/orderKeywordSearch.js:72]: I also foresee the opportunity to put some sugar in here -- e.g. `const results = sweetFunc(queryMatches);` that does all these things and also brings back the `_score`

File: shared/src/business/useCaseHelper/search/orderSearch.test.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
+const {
+  applicationContext,
+} = require('../../test/createTestApplicationContext');
+const { map } = require('lodash');
+const { Order } = require('../../entities/orders/Order');
+const { orderKeywordSearch } = require('./orderKeywordSearch');
+
+describe('orderKeywordSearch', () => {
+  beforeEach(() => {
```
Comment (kkoskelin) [inline shared/src/business/useCaseHelper/search/orderSearch.test.js]: can be `beforeAll`

File: web-api/src/public-api/orderPublicSearchLambda.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const { genericHandler } = require('../genericHandler');
+
+/**
+ * used for fetching cases matching the given name, country, state, and/or year filed range for the general public
```
Comment (kkoskelin) [inline web-api/src/public-api/orderPublicSearchLambda.js]: this is searching orders, not fetching cases -- right?

**Review by kkoskelin — CHANGES_REQUESTED** [review]
Only thing worth holding up are some jsdoc adjustments -- but I can't wait to get some better practices around the ES queries and results handling, too.

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**